tidy3d/components/simulation.py

```diff
@@ -4540,8 +4641,13 @@ def nyquist_step(self) -> int:
         # combined frequency upper bound
         freq_max = max(freq_source_max, freq_monitor_max)
 
+        # in fixed angle simulations both E and H are available at full and half steps
+        fixed_angle_factor = 1
+        if len(self._fixed_angle_sources) > 0:
+            fixed_angle_factor = 2
+
         if freq_max > 0:
-            nyquist_step = int(1 / (2 * freq_max) / self.dt) - 1
+            nyquist_step = int(1 / (2 * freq_max) / self.dt * fixed_angle_factor) - 1
```

Comment:
As usual you've thought of everything :) Did this manifest as an issue that you had to debug to find out or you just thought about it in the process of developing?

Reply:
it was naturally following from backend changes